src/water_cmd.cpp

```diff
@@ -477,8 +477,9 @@ CommandCost CmdBuildCanal(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
 
 		bool water = IsWaterTile(current_tile);
 
-		/* can't make water of water! */
-		if (water && (!IsTileOwner(current_tile, OWNER_WATER) || wc == WATER_CLASS_SEA)) continue;
+		/* Allow to make water of water, except in these specific situations.
+		 * The checks done via CMD_LANDSCAPE_CLEAR can provide better error messages if any. */
```

Comment:
This comment doesn't really help the reader. "These specific situations" is what is important. As the code is not obvious in what those situations are.
Also, the "make water of water" doesn't make sense anymore.

You have to look at it this way: if you read this in 1 year, would you still understand what this code does? And if not, will the comment help you understanding?

So what are those specific situations? Well, if I understood you correctly, your intentions are:
- Building a canal on your own canal should be skipped.
- Building a canal on a `OWNER_NONE` owned canal should be skipped.

So the comment should at least mention that.

Basically, what you made this `if` do now, is something like:
`Prevent building a canal over your own or OWNER_NONE owned canal`.